go/cmd/dolt/commands/reflog.go

```diff
@@ -0,0 +1,206 @@
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
+	query, err := constructInterpolatedDoltReflogQuery(apr)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	rows, err := GetRowsForSql(queryist, sqlCtx, query)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	return printReflog(rows, queryist, sqlCtx)
+}
+
+// constructInterpolatedDoltReflogQuery generates the sql query necessary to call the DOLT_REFLOG() function.
+// Also interpolates this query to prevent sql injection
+func constructInterpolatedDoltReflogQuery(apr *argparser.ArgParseResults) (string, error) {
+	var params []interface{}
+	refPlaceholder := ""
+	allFlag := ""
+
+	if apr.NArg() == 1 {
+		params = append(params, apr.Arg(0))
+		refPlaceholder = "?"
+	}
+	if apr.Contains(cli.AllFlag) {
+		allFlag = "'--all'"
+	}
+
+	args := ""
+	if refPlaceholder == "" && allFlag != "" {
+		args = allFlag
+	} else if refPlaceholder != "" && allFlag == "" {
+		args = refPlaceholder
+	} else if refPlaceholder != "" && allFlag != "" {
+		args = strings.Join([]string{refPlaceholder, allFlag}, ", ")
+	}
+
+	query := strings.Join([]string{"SELECT ref, commit_hash, commit_message FROM DOLT_REFLOG(", args, ")"}, "")
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		return "", err
+	}
+
+	return interpolatedQuery, nil
+}
+
+type ReflogInfo struct {
+	ref           string
+	commitHash    string
+	commitMessage string
+}
+
+// printReflog takes a list of sql rows with columns ref, commit hash, commit message. Prints the reflog to stdout
+func printReflog(rows []sql.Row, queryist cli.Queryist, sqlCtx *sql.Context) int {
+	var reflogInfo []ReflogInfo
+
+	// Get the current branch
+	res, err := GetRowsForSql(queryist, sqlCtx, "SELECT active_branch()")
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), nil)
+	}
+	curBranch := res[0][0].(string)
+
+	for _, row := range rows {
+		ref := row[0].(string)
+		commitHash := row[1].(string)
+		commitMessage := row[2].(string)
+		reflogInfo = append(reflogInfo, ReflogInfo{ref, commitHash, commitMessage})
+	}
+
+	reflogToStdOut(reflogInfo, curBranch)
+
+	return 0
+}
+
+// reflogToStdOut takes a list of ReflogInfo and prints the reflog to stdout
+func reflogToStdOut(reflogInfo []ReflogInfo, curBranch string) {
+	if cli.ExecuteWithStdioRestored == nil {
+		return
+	}
+	cli.ExecuteWithStdioRestored(func() {
+		pager := outputpager.Start()
+		defer pager.Stop()
+
+		for pos, info := range reflogInfo {
+			// TODO: use short hash instead
+			line := []string{fmt.Sprintf("\033[33m%s\033[0m", info.commitHash)} // commit hash in yellow (33m)
+
+			processedRef := processRefForReflog(info.ref, curBranch)
+			line = append(line, fmt.Sprintf("\033[33m(%s\033[33m)\033[0m", processedRef)) // () in yellow (33m)
+			line = append(line, fmt.Sprintf("HEAD@{%d}: %s\n", pos, info.commitMessage))
```

Comment:
I don't think we should include the `HEAD@{%d}` syntax... In Git, those are actually usable ways to reference commits, but we don't support that syntax, so it's not really doing anything helpful. Also, the name should be the name of the ref who's log you're searching. So... in Git, by default when you run `git reflog`, you are looking at the special `HEAD` reference (which we don't support in Dolt), so you see `HEAD@{1}`, etc. For example, when I select a different reflog to view with Git, by running `git reflog refs/heads/main`, I'll see `refs/heads/main@{11}` instead of `HEAD@{11}`. 

Another reason not to show these is because Dolt's reflog works a little differently than Gits – it allows you to see all the reflogs interwoven together with the `--all` flag, it would be difficult to make those labels meaningful/correct and even if they did match the ref and use the correct ID, they still wouldn't be useable for anything, so it just feels like noise that will get in customers' way.